### PR TITLE
Create POST endpoint for books model

### PR DESCRIPTION
**Pull request by spreddy1717**
https://whitepip.atlassian.net/browse/WP-5

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -6,6 +6,7 @@ name = "pypi"
 [packages]
 django = "==4.2.4"
 djangorestframework = "*"
```
Comment (srinivasreddy) [inline Pipfile]: pin it down to latest long term supported version.

Comment (srinivasreddy) [reply Pipfile]: `pipenv install djangorestframework==3.14.0`

File: Pipfile [hunk 2]
```diff
@@ -6,6 +6,7 @@ name = "pypi"
...
+pillow = "*"
```
Comment (srinivasreddy) [inline Pipfile]: Let us pin down the latest stable version. 

`s/pillow = '*'/ pillow = "==10.0.0"`

File: Pipfile (hunk 1 above)
Comment (srinivasreddy) [reply Pipfile]: change the version to 3.14.0

File: books/models.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
+from django.db import models
+from django.contrib.auth.models import User
+
+
+class Author(models.Model):
+    uuid = models.UUIDField(primary_key=True)
```
Comment (srinivasreddy) [inline books/models.py]: Implement customer uuid for author starting with `author_`

File: books/models.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    biography = models.TextField(blank=True, null=True)
+    date_of_birth = models.DateField(blank=True, null=True)
+    profile_picture = models.ImageField(upload_to="profile/", blank=True, null=True)
+
+    def __str__(self):
+        return self.name
+
+
+class Category(models.Model):
+    uuid = models.UUIDField(primary_key=True)
```
Comment (srinivasreddy) [inline books/models.py]: similar as Author model

File: books/models.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    Categeoryname = models.CharField(max_length=50)
```
Comment (srinivasreddy) [inline books/models.py]: always use snake  case for column names.

File: books/models.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    description = models.TextField(max_length=1000)
+
+    def __str__(self):
+        return self.Categeoryname
+
+
+class Book(models.Model):
+    uuid = models.UUIDField(primary_key=True)
```
Comment (srinivasreddy) [inline books/models.py]: similar as Author and Category models.

File: books/models.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    title = models.CharField(max_length=1000)
+    ISBN = models.CharField(max_length=13, unique=True)
```
Comment (srinivasreddy) [inline books/models.py]: lower case pls

File: books/models.py [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    summary = models.TextField()
+    price = models.DecimalField(decimal_places=2, max_digits=7)
```
Comment (srinivasreddy) [inline books/models.py:41]: The maximum price of a book can be 1 lakhs ( 5 digits)  and 99 (2 digits) paise. Total 7 digits.

File: Pipfile (hunk 1 above)
Comment (spreddy1717) [reply Pipfile]: changed it,but why is it not showing updated ,even the pushes i made were not kicked in

File: books/urls.py [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+urlpatterns = [
+    path(
+        "swagger/",
+        schema_view.with_ui("swagger", cache_timeout=0),
+        name="schema-swagger-ui",
+    ),
+    path("api/books/", BookCreateView.as_view(), name="book-create"),
+    path("api/books/list/", BookListView.as_view(), name="book-list"),
+    path("api/books/<uuid:uuid>/", BookGetDeleteView.as_view(), name="book-delete"),
```
Comment (srinivasreddy) [inline books/urls.py]:
```
path("api/books/<uuid:uuid>/", BookGetDeleteView.as_view(), name="book-delete"),
path("api/books/<uuid:pk>/", UpdateBook.as_view(), name="update-book")
```
Combine the above two into a single-view class
```
path("api/books/<str:pk>/", BookGetUpdateDeleteView.as_view(), name="book-get-update-delete"),
```

Comment (srinivasreddy) [reply books/urls.py]: Apply the same logic for `authors` and `category` too.  💯 💯

**Review by srinivasreddy — APPROVED** [review]
Approve

**Pull request merged**